waterlog/src/actions/MapActions.js

```diff
@@ -51,18 +51,62 @@ async function getData(dispatch){
     dispatch(fetchMapsDataFailure(error));
   });
 
-  return {segments, monitors, leaks}
+  return { segments, monitors, leaks }
 }
 
-export const fetchMapsData = () => (dispatch) => {
-	dispatch(fetchMapsDataBegin());
+function formatMapData(data) {
+  let markers, segments;
+  let todaysLeaks;
+  let date = new Date(Date.now());
+  if (!data.leaks) {
+    todaysLeaks = [];
+  } else {
+    todaysLeaks = data.leaks.filter(
+      leak =>
+        new Date(leak.latestTimeStamp).getDay() === date.getDay() &&
```

Comment:
You basically calling the function three times and another function on top of that. 
Extract `new Date(leak.latestTimeStamp)` into a variable and then it will at least only be 
`variableName.getDay()`

Reply:
I changed that to using one variable